docs/apps/molpro.md

```diff
@@ -9,62 +9,90 @@ MOLPRO is a software package geared towards accurate ab initio quantum chemistry
-#SBATCH --time=00:10:00           # time as `hh:mm:ss`
-
-module load molpro/2024.1
-
-export MOLPRO_TMP=$PWD/MOLPRO_TMP_$SLURM_JOB_ID
-mkdir -p $MOLPRO_TMP
 
-$MOLPROP -d$MOLPRO_TMP -I$MOLPRO_TMP -W$PWD test.com
-rm -rf $MOLPRO_TMP
-```
+Molpro has been built with the Global Arrays toolkit (`--with-mpi-pr`) that allocates one helper process per node for parallel MPI runs.
 
 !!! info "Note"
-    Particularly some of the wavefunction-based electron correlation methods can be very disk I/O intensive. Such jobs benefit from using the [fast local storage](../../computing/running/creating-job-scripts-puhti/#local-storage) on Puhti. Using local disk for such jobs will also reduce the load on the Lustre parallel file system.
+    Although some parts of the code support shared memory parallelism (OpenMP), its use is not generally recommended.
 
-### Example batch script for Puhti using MPI parallelization and local disk (NVMe)
+### Example batch scripts
 
-```bash
-#!/bin/bash
-#SBATCH --partition=small
-#SBATCH --nodes=1
-#SBATCH --ntasks-per-node=40 # MPI tasks per node
-#SBATCH --account=<project>  # insert here the project to be billed 
-#SBATCH --time=00:10:00      # time as `hh:mm:ss`
-#SBATCH --gres=nvme:100      # requested local disk space in GB 
-
-module load molpro/2024.1
-export MOLPRO_TMP=$LOCAL_SCRATCH/$SLURM_JOB_ID
-mkdir -p $MOLPRO_TMP
-
-$MOLPROP -d$MOLPRO_TMP -I$MOLPRO_TMP -W$PWD test.com
-rm -rf $MOLPRO_TMP
-```
+!!! info "Note"
+    Wave function-based correlations methods, both single and multireference, often create a
+    substantial amount of disk I/O. In order to achieve maximal performance for the job and to
+    avoid excess load on the Lustre parallel file system it is advisable to use the local disk.
+
+=== "Puhti"
+
+    ```bash
+    #!/bin/bash
+    #SBATCH --partition=test
+    #SBATCH --nodes=2
+    #SBATCH --ntasks-per-node=40      # MPI tasks per node
+    #SBATCH --account=yourproject     # insert here the project to be billed 
+    #SBATCH --time=00:15:00           # time as `hh:mm:ss`
+    module purge
+    module load molpro/2024.3
+
+    export MOLPRO_TMP=$PWD/MOLPRO_TMP_$SLURM_JOB_ID
+    mkdir -p $MOLPRO_TMP
+
+    $MOLPROP -d$MOLPRO_TMP -I$MOLPRO_TMP -W$PWD test.com
+    rm -rf $MOLPRO_TMP
+    ```
+
+=== "Puhti, local disk"
+
+    ```bash
+    #!/bin/bash
+    #SBATCH --partition=large
+    #SBATCH --nodes=2
+    #SBATCH --ntasks-per-node=40
+    #SBATCH --account=yourproject     # insert here the project to be billed
+    #SBATCH --time=00:15:00           # time as `hh:mm:ss`
+    #SBATCH --gres=nvme:100           # requested local disk space in GB
+    module purge
+    module load molpro/2024.3
+    export MOLPRO_TMP=$LOCAL_SCRATCH/MOLPRO_TMP_$SLURM_JOB_ID
+    mkdir -p $MOLPRO_TMP
+
+    $MOLPROP -d$MOLPRO_TMP -I$MOLPRO_TMP -W$PWD test.com
+    rm -rf $MOLPRO_TMP
+    ```
+
+=== "Mahti"
+
+    ```bash
+    #!/bin/bash
+    #SBATCH --partition=test
+    #SBATCH --nodes=1
+    #SBATCH --ntasks-per-node=16
```

Comment:
Should the Mahti example use a full node to make clear that sub-node allocations are generally not possible (only on interactive partition).

Reply:
In practice, most Molpro calculations on Mahti require undersubscribing cores to ensure sufficient memory per core. Should this be clarified in the example?